cips/cip-19.md

```diff
@@ -311,6 +311,49 @@ Namespace data may span over multiple rows, in which case all the data is encaps
 containers. This enables parallelization of namespace data retrieval and certain [compositions](#protocol-compositions)
 may get advantage of that by requesting containers of a single namespace from multiple servers simultaneously.
 
+#### NamespaceRangeID
+
+NamespaceRangeID: encapsulates SampleID, Namespace and identifies the continuous range of shares in the DataSquare. Formated as below:
+
+```text
+NamespaceRangeID {
+  SampleID;
```

Comment:
what is the purpose of sample here? does it mean take `Length` shares from given `SampleID`? I think it needs better clarification below. Also why not just using `EdsId` + `Namespace` providing whole namespace from given block? Is there a usecase for a user to request his shares but eg. skipping the first and last one?

Reply:
> does it mean take Length shares from given SampleID?

Yes

Reply:
> Also why not just using EdsId + Namespace providing whole namespace from given block? Is there a usecase for a user to request his shares but eg. skipping the first and last one?

This means we will retrieve the whole eds from the requested namespace. For example, `blob.GetProof` needs a specific range of shares rather than the whole namespace.

Reply:
should we then specify here what should happen if cid is pointing to shares outside of it's namespace? eg. what if the sample id is not in correct namespace or the length goes beyond the namespace?

Reply:
also I'm not exactly sure what is the usecase here, user would need to already know the blob's position in eds and it's length in shares to even use that, however shwap itself doesn't give any of such abilities